edb/schema/indexes.py

```diff
@@ -1244,6 +1265,80 @@ def validate_object(
+        if (
+            referrer_ctx is not None
+            and not context.canonical
+            and (ai_index := schema.get(
+                sn.QualName("ext::ai", "index"),
+                type=Index,
+                default=None,
+            )) is not None
+            and self.scls.issubclass(schema, ai_index)
+        ):
+            kwargs = self.scls.get_concrete_kwargs(schema)
+            model_name_expr = kwargs["embedding_model"]
+            model_name = model_name_expr.assert_compiled().as_python_value()
+
+            models = get_defined_ext_ai_embedding_models(schema, model_name)
+            if len(models) == 0:
+                raise errors.SchemaDefinitionError(
+                    f'undefined embedding model: no subtype of '
+                    f'ext::ai::EmbeddingModel is annotated as {model_name!r}',
+                    span=self.span,
+                )
+            elif len(models) > 1:
+                models_dn = [
+                    model.get_displayname(schema) for model in models.values()
+                ]
+                raise errors.SchemaDefinitionError(
+                    f'expecting only one embedding model to be annotated '
+                    f'with ext::ai::model_name={model_name!r}: got multiple: '
+                    f'{", ".join(models_dn)}',
+                    span=self.span,
+                )
+
+            model_stype = next(iter(models.values()))
+            model_stype_vn = model_stype.get_verbosename(schema)
+            model_annos = model_stype.get_annotations(schema)
+            my_name = self.scls.get_name(schema)
+            for model_anno in model_annos.objects(schema):
+                anno_name = model_anno.get_shortname(schema)
+                if anno_name.module != "ext::ai":
+                    continue
+                value = model_anno.get_value(schema)
+                if value is None or value == "<must override>":
+                    raise errors.SchemaDefinitionError(
+                        f"{model_stype_vn!r} is missing a value for the "
+                        f"{anno_name!r} annotation"
+                    )
+                anno_sname = sn.get_specialized_name(
+                    anno_name,
+                    str(my_name),
+                )
+                schema1 = model_anno.update(
+                    schema,
+                    {
+                        "name": sn.QualName(my_name.module, anno_sname),
+                        "subject": self.scls,
+                    },
+                )
+                anno_copy = schema1.get(
+                    model_anno.get_name(schema1),
+                    type=s_anno.AnnotationValue,
+                )
+                anno_cmd = anno_copy.as_create_delta(
+                    schema1, so.ComparisonContext())
+                anno_cmd.discard_attribute("bases")
+                anno_cmd.discard_attribute("ancestors")
+                self.add(anno_cmd)
```

Comment:
Yeah, we are essentially copying the relevant annotations from the declaration of the model specified in the `embedding_model` index kwarg.  This is needed because looking those annotations up from the model at run time would be very inefficient.

Bases are cleared because we don't want to "inherit" the annotations from the abstract base model's definition.